drivers/syslog/syslog_filechannel.c

```diff
@@ -85,28 +89,44 @@ static void log_rotate(FAR const char *log_file)
       return;
     }
 
-  /* Construct the backup file name. */
+  /* Rotated file names. */
 
-  backup_file = malloc(strlen(log_file) + 3);
-  if (backup_file == NULL)
+  name_size = strlen(log_file) + 8;
+  rotate_to = malloc(name_size);
+  rotate_from = malloc(name_size);
+  if ((rotate_to == NULL) || (rotate_from == NULL))
     {
-      return;
+      goto end;
     }
 
-  sprintf(backup_file, "%s.0", log_file);
+  /* Rotate the logs. */
+
+  for (i = (CONFIG_SYSLOG_FILE_ROTATIONS - 1); i > 0; i--)
+    {
+      snprintf(rotate_to, name_size, "%s.%d", log_file, i);
+      snprintf(rotate_from, name_size, "%s.%d", log_file, i - 1);
+
+      if (access(rotate_to, F_OK) == 0)
+        {
+          remove(rotate_to);
+        }
+
+      rename(rotate_from, rotate_to);
```

Comment:
should we check:
if (access(rotate_from, F_OK) == 0)
  {
    rename(rotate_from, rotate_to);
  }

Reply:
Why spend time on this check? `rename` will check whether `rotate_from` exists.

And if it fails, there are not much we can do. Nor it affects us in any way.